### PR TITLE
Add types of review section

### PR DESCRIPTION
**Pull request by joshwlambert**
This PR adds a section to the Code review (`code-review.qmd`) chapter of blueprints on best practises for the types of reviews conducted in Epiverse-TRACE.

### CODE REVIEW

File: code-review.qmd [hunk 1]
```diff
@@ -36,3 +36,60 @@ However, in a scenario where the suggested change cannot be commited to the feat
...
+
+The full package review requires a similar setup process to the version release partial code review. Instead of a branch from the previous release, we want a branch that is completely empty. This is not possible, but we can create the next best thing, a branch from the first commit using:
+
+```sh
+git rev-list --all | tail -1
+git branch empty <commit_sha>
+git push origin empty
+```
+
+This retrieve the commit SHA and then creates and pushes the branch. Then the `review` branch can be created with the same method as the partial review:
```
Comment (jamesmbaazam) [inline code-review.qmd:77]:
```suggestion
This retrieves the commit SHA and then creates and pushes the branch. Then the `review` branch can be created with the same method as the partial review:
```

**Review by jamesmbaazam — APPROVED** [review]

**Comment by jamesmbaazam** [comment]
Thanks for writing this @joshwlambert. Very insightful. Based on our previous conversations, I thought we referred to reviews of small code additions to the code base as "partial reviews" and reviews of the whole code base at any time as full package reviews. I think that distinction is clear to me but the definition of "partial" here could potentially be confusing. I interpret it to mean that the first full review before the first release of a package is the only full review. Is that the definition we're going with? If we were to go with my definition, then this article could easily be restructured to describe the two ways to do full package reviews: one from the first commit, and one from any commit, where here, we refer to the commit linked to the last release. 

What are your thoughts?

**Comment by joshwlambert** [comment]
@jamesmbaazam I'm not sure I understand your comment. 

> I thought we referred to reviews of small code additions to the code base as "partial reviews" and reviews of the whole code base at any time as full package reviews

This is the definition I've used when writing the section.

> I interpret it to mean that a full review only happens when you compare the state of the code base to the initial commit.

Yes, this is the definition I am going with and have been assuming other people are using. A full review can happen locally (i.e. cloning the repo and reviewing all the code) at any time but, without the full package PR, GitHub does not provide all differences on the Files changed tab of the PR.

**Comment by jamesmbaazam** [comment]
> but, without the full package PR, GitHub does not provide all differences on the Files changed tab of the PR.

This also happens with the approach described in the second paragraph of the partial review section, doesn't it? Wouldn't that therefore be considered a full package (code base) review as well since the reviewer can now comment on all aspects of the code base as if they were touched?

**Comment by joshwlambert** [comment]
No that is not the case, the partial review between versions only provides the files that have changed since the last version release. In essence it's just a slightly more involved process of creating a standard PR that you would make from a feature branch to `main`.

**Comment by jamesmbaazam** [comment]
> * Rather than making the "targeting first commit" vs "targeting a previous, but not the first commit" distinction, I would rather make a distinction between "pre-release review" / "regular large scale review" / another name and "new feature or bug fix review". Since these reviews should be approached differently and with a different level of scrutiny IMO. In other words, [{superspreading} v0.2.0 pkg review superspreading#77](https://github.com/epiverse-trace/superspreading/pull/77) and [{superspreading} v.0.1.0 full pkg review superspreading#31](https://github.com/epiverse-trace/superspreading/pull/31) belong to the same category for me. If I understand correctly, this was also what James was saying.

Thanks, @Bisaloo. That is what I was saying, but upon further discussion with Josh, I think it would be better to keep the current sections, i.e., full review (the whole code base can be review or commented on) vs partial reviews (review of the current state to an incoming change or previous state), where we can have subsections like reviews of new feature, two states of the package (one version vs another), documentation reviews, etc. The current sections are good because they don't place any conditions on what happens after the review. A "pre-release" review would suggest that there will be a release after the review, but it could just be a routine full review to refactor the code base.

**Comment by joshwlambert** [comment]
Thanks both for the suggestions. I have made a few minor changes. I don't have anything else to commit, this PR can be merged when ready.

**Review by Bisaloo — APPROVED** [review]

**Pull request merged**